server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```diff
@@ -437,8 +437,8 @@ public boolean shouldPeriodicallyFlush() {
     }
 
     @Override
-    public void flush(boolean force, boolean waitIfOngoing) throws EngineException {
-        // noop
+    public boolean flush(boolean force, boolean waitIfOngoing) throws EngineException {
+        return true; // noop
```

Comment:
I think this should return false? This implementation won't flush.

Reply:
But that is not really the meaning of the return value as it stands in `InternalEngine`. `InternalEngine` may also return true without flushing. What the return value currently mean is

false: if `waitIfOngoing` detected a collision
true: otherwise.

Here true seems most correct since no collision was detected.

Reply:
Yes, I was wrong, please dismiss my comment.